# Q1

In [20]:
import pandas as pd
import numpy as np
from scipy.stats import chi2_contingency
from scipy.stats import chi2

In [4]:
cutlets_df = pd.read_csv("Cutlets.csv")
cutlets_df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [11]:
x̄ = cutlets_df['Unit A'].mean()
x̄

7.01909142857143

In [12]:
ȳ = cutlets_df['Unit B'].mean()
ȳ

6.964297142857142

# Null Hypothesis is µ1=µ2, Alternate Hypothesis is µ1!=µ2

In [13]:
s1 = cutlets_df['Unit A'].std()
s1

0.2884084841815496

In [14]:
s2 = cutlets_df['Unit B'].std()
s2

0.343400647063108

In [22]:
n = cutlets_df.shape[0]
n

35

In [23]:
t = (x̄ - ȳ)/np.sqrt((s1**2 + s2**2)/n)
t

0.7228688704678298

# Since value of t lies between -2 and +2 null hypothesis should be accepted that is average diameter of both units is same

# Q2

In [2]:
labtats_df = pd.read_csv("LabTAT.csv")
labtats_df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [3]:
import scipy.stats as stats
stats.f_oneway(labtats_df.iloc[:,0], labtats_df.iloc[:,1],labtats_df.iloc[:,2],labtats_df.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

# Null Hypothesis is avg TAT of 4 labs is same

# p is approximately 0 and α = 0.05 since p is less than α we reject the null hypothesis. Hence average turn around time of the 4 laboratories is different

# Q3

In [34]:
buyers_df = pd.read_csv("BuyerRatio.csv")
buyers_df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [37]:
buyers_df = buyers_df.iloc[:,1:5]
buyers_df

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [39]:
chisquare = chi2_contingency(buyers_df)
chisquare

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [44]:
test_statistic = chisquare[0]
test_statistic

1.595945538661058

In [42]:
df = chisquare[2]
df

3

In [40]:
α = 0.05
α

0.05

In [41]:
prob = 1-α
prob

0.95

In [43]:
critical_value = chi2.ppf(prob, df)
critical_value

7.814727903251179

# test statistic is 1.6 and critical value is 7.8. Since test statistic is less than critical value we fail to reject null hypothesis that is we accept null hypothesis

# Q4

In [5]:
orders_df = pd.read_csv("Customer_OrderForm.csv")
orders_df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [16]:
print(orders_df['Phillippines'].value_counts(),'\n')
print(orders_df['Indonesia'].value_counts(),'\n')
print(orders_df['Malta'].value_counts(),'\n')
print(orders_df['India'].value_counts(),'\n')

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 

Error Free    267
Defective      33
Name: Indonesia, dtype: int64 

Error Free    269
Defective      31
Name: Malta, dtype: int64 

Error Free    280
Defective      20
Name: India, dtype: int64 



In [17]:
observed = [[271,267,269,280],[29,33,31,20]]
observed

[[271, 267, 269, 280], [29, 33, 31, 20]]

In [19]:
chisquare = chi2_contingency(observed)
chisquare

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [24]:
α = 0.05
α

0.05

In [28]:
df = chisquare[2]
df

3

In [26]:
prob = 1-α
prob

0.95

In [30]:
test_statistic = chisquare[0]
test_statistic

3.858960685820355

In [31]:
critical_value = chi2.ppf(prob, df)
critical_value

7.814727903251179

# Null hypothesis is defective percentage is same across all centers
# Alternate hypothesis is defective percentage varies across different centers

# critical value is 7.815. Since the test-statistic is 3.86 which is smaller than the critical value of 7.815
# We cannot reject the null hypothesis that defective percentage is same across all centers. Therefore defective percentage is same across all centers.